### PR TITLE
Feature/mat/ci

### PR DESCRIPTION
**Pull request by m-mcgowan**
Adds CI build using headless invocation of STM32CubeIDE.  Builds all configurations in the project - Debug and Release.

WiP/PoC for review and incremental adjustment.

Some improvements:

* presently using a docker image based on STM32CubeIDE 1.4.0.  More recent versions are available, such as this one, https://github.com/xanderhendriks/docker-stm32cubeide

* The repo presently includes the output build directory `Debug`.  We can move these to github releases.  Automated release builds via an action that creates a release when a tagged build is done performed.

### CODE REVIEW

File: ci.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# echo commands
+set -x
+
+
+
+# This is generic (so why not put in the docker image?)
+
+STM32CUBEIDE_PATH=${STM32CUBEIDE_PATH:-/opt/st/stm32cubeide_1.4.0}
```
Comment (zfields) [inline ci.sh:10]: The version of the STM32CubeIDE seems a little long in the the tooth, but doesn't seem to be problematic :shrug:. I imagine it's baked into the STM32 action you are using, right?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+on: [push, pull_request]
+
+jobs:
+  stm32_ci_job:
+    runs-on: ubuntu-latest
+    name: CI job for stm32
+    steps:
+    - uses: actions/checkout@v2
```
Comment (zfields) [inline .github/workflows/main.yml]: This is a bit terse, but the YAML syntax (`-`) makes it clear I suppose. I had to do a double take not to confuse it with the step below.

Our other CI workflows are more explicit, like...
```suggestion
    - name: Checkout Code
      id: checkout
      uses: actions/checkout@v2
      
```

**Review by zfields — APPROVED** [review]
Clever little action - looks neat! I think it's the appropriate amount of effort for the time being. :+1: 

Maybe someday we roll our own, with a newer version of STM32CubeIDE (if/when we ever need to).

File: ci.sh (hunk 1 above)
Comment (zfields) [reply ci.sh:10]: I didn't read your first message... :facepalm: I'll just reiterate, I don't think it's worth changing at this time.

File: .github/workflows/main.yml (hunk 2 above)
Comment (m-mcgowan) [reply .github/workflows/main.yml]: Right, I like the explicit naming better. I didn't quite grok it when I saw the yaml so thanks for highlighting it.

**Comment by m-mcgowan** [comment]
[Development Examples: remove hard-coded PRODUCT_UIDs from sample apps and example repos.](https://trello.com/c/gXVvrnf3/41-development-examples-remove-hard-coded-productuids-from-sample-apps-and-example-repos)

**Pull request merged**